### PR TITLE
dev-util/bitrise: new ebuild, version 1.16.1

### PR DESCRIPTION
**Pull request by koral--**
Bitrise is a Continuous Integration and Delivery (CI/CD) Platform as a Service (PaaS) with a main focus on mobile app development (iOS, Android, Xamarin, ...).

Bug: https://bugs.gentoo.org/655770
Closes: https://bugs.gentoo.org/655770

### CODE REVIEW

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+DESCRIPTION="Run your Bitrise.io automations on any Mac or Linux machine."
+HOMEPAGE="https://www.bitrise.io/cli"
+SRC_URI="${ARCHIVE_URI}"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="amd64"
+IUSE="doc"
+
+DEPEND=""
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: drop it

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+RDEPEND=""
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: drop it

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+src_compile() {
+	GOPATH="${WORKDIR}" go build -v -o "${BITRISE_BIN}" || die "Couldn't compile bitrise"
+
+}
+
+src_test() {
+	export PULL_REQUEST_ID=""
+	export INTEGRATION_TEST_BINARY_PATH="${S}/${BITRISE_BIN}"
+	rm "${S}/_tests/integration/version_test.go"
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: `|| die`

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	GOPATH="${WORKDIR}" go test -v ./_tests/integration/... || die "Integration tests failed"
+}
+
+src_install() {
+	dobin "${BITRISE_BIN}" || die "Couldn't install bitrise binary"
+	if use doc; then
+		dodoc -r _docs README.md || die "Couldn't install documentation"
+	fi
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: no `|| die`

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+src_test() {
+	export PULL_REQUEST_ID=""
+	export INTEGRATION_TEST_BINARY_PATH="${S}/${BITRISE_BIN}"
+	rm "${S}/_tests/integration/version_test.go"
+	GOPATH="${WORKDIR}" go test -v ./_tests/integration/... || die "Integration tests failed"
+}
+
+src_install() {
+	dobin "${BITRISE_BIN}" || die "Couldn't install bitrise binary"
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: `|| no die`

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+	if use doc; then
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: `use doc && <condition>`

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+EAPI=6
+
+inherit golang-build
+
+EGO_ON="github.com/bitrise-io"
+EGO_PN="${EGO_ON}/${PN}"
+ARCHIVE_URI="https://${EGO_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+DESCRIPTION="Run your Bitrise.io automations on any Mac or Linux machine."
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: no dot in the end

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+HOMEPAGE="https://www.bitrise.io/cli"
+SRC_URI="${ARCHIVE_URI}"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="amd64"
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: not straight to stable

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,50 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit golang-build
+
+EGO_ON="github.com/bitrise-io"
+EGO_PN="${EGO_ON}/${PN}"
+ARCHIVE_URI="https://${EGO_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: why do it more complicated?

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+DESCRIPTION="Run your Bitrise.io automations on any Mac or Linux machine."
+HOMEPAGE="https://www.bitrise.io/cli"
+SRC_URI="${ARCHIVE_URI}"
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: see above ^

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+DEPEND=""
+RDEPEND=""
+
+S="${WORKDIR}/src/${EGO_ON}/${PN}"
+BITRISE_BIN="bin/${PN}"
+
+src_unpack() {
+	unpack ${A}
+	mkdir -p "${WORKDIR}/src/${EGO_ON}"
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: `|| die`

File: dev-util/bitrise/metadata.xml [hunk 12]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Zlogene) [inline dev-util/bitrise/metadata.xml]: add gentoo-proxy-maint project last, as you are not a developer

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 13]
```diff
@@ -7,44 +7,37 @@ inherit golang-build
 
 EGO_ON="github.com/bitrise-io"
 EGO_PN="${EGO_ON}/${PN}"
-ARCHIVE_URI="https://${EGO_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
 
-DESCRIPTION="Run your Bitrise.io automations on any Mac or Linux machine."
+DESCRIPTION="Run your Bitrise.io automations on any Mac or Linux machine"
 HOMEPAGE="https://www.bitrise.io/cli"
-SRC_URI="${ARCHIVE_URI}"
+SRC_URI="https://${EGO_PN}/archive/${PV}.tar.gz"
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: it gives you wrong manifest, so `-> ${P}.tar.gz`

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 14]
```diff
@@ -7,44 +7,37 @@ inherit golang-build
...
+	rm "${S}/_tests/integration/version_test.go" || die "Couldn't prepare integration tests"
 	GOPATH="${WORKDIR}" go test -v ./_tests/integration/... || die "Integration tests failed"
 }
 
 src_install() {
-	dobin "${BITRISE_BIN}" || die "Couldn't install bitrise binary"
-	if use doc; then
-		dodoc -r _docs README.md || die "Couldn't install documentation"
-	fi
+	dobin "${BITRISE_BIN}"
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: why did you intriduce actually useless shortcuts for path? there is no reason for this, just give the full path `bin/bitrise`

**Comment by Zlogene** [comment]
Also please always squash your commits into the single one

File: dev-util/bitrise/bitrise-1.16.0.ebuild [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+HOMEPAGE="https://www.bitrise.io/cli"
+SRC_URI="https://${EGO_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="doc"
+
+S="${WORKDIR}/src/${EGO_ON}/${PN}"
+BITRISE_BIN="bin/${PN}"
```
Comment (Zlogene) [inline dev-util/bitrise/bitrise-1.16.0.ebuild]: And do you actually need this variable now, then?

**Comment by koral--** [comment]
@Zlogene I've applied all the requested changes.

File: dev-util/bitrise/bitrise-1.16.1.ebuild [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+
+src_test() {
+	export PULL_REQUEST_ID=""
+	export INTEGRATION_TEST_BINARY_PATH="${S}/bin/bitrise"
+	GOPATH="${WORKDIR}" go test -v ./_tests/integration/... || die "Integration tests failed"
+}
+
+src_install() {
+	dobin bin/bitrise
+	use doc && dodoc -r _docs README.md
```
Comment (mgorny) [inline dev-util/bitrise/bitrise-1.16.1.ebuild]: Please install `README.md` unconditionally.

**Pull request closed without merging**